core/DataAccess/ArchiveSelector.php

```diff
@@ -163,7 +163,6 @@ public static function getArchiveIds($siteIds, $periods, $segment, $plugins, $in
                                FROM %s
                               WHERE idsite IN (" . implode(',', $siteIds) . ")
                                 AND " . self::getNameCondition($plugins, $segment, $includeInvalidated) . "
-                                AND ts_archived IS NOT NULL
```

Comment:
Did you check if removing that condition might have any side effects? Looking thought the code it seems an empty `ts_archived` might mean that the archive is "temporary" or not yet finished. But not sure if that is still the case, as I couldn't find anything where an empty ts_archived might be inserted.

Reply:
@sgiehl I've looked through the entire codebase and couldn't find anything that ever unsets this value. I know it was added just 2.5 years ago in https://github.com/matomo-org/matomo/commit/6d744db7f3f7a55fc4bb1ca573e937f0832c642b but I can't find anything around it. Neither when i search for terms like "placeholder". This would also indicate that https://github.com/matomo-org/matomo/blob/4.13.0/core/Archive/ArchivePurger.php#L102-L103 is unneeded.

I've looked through our various DBs and couldn't find any `ts_archived is null` value in recent archives.

I also searched through logs for this message https://github.com/matomo-org/matomo/blob/4.13.0/core/Archive/ArchivePurger.php#L105 to see if we ever purged entries, but the problem is that the count of archiveIds is not only from the `getPlaceholderArchiveIds()` method but also includes archives from `getInvalidatedArchiveIdsSafeToDelete`. I will temporarily add for few hours some extra logging there to double check if we ever delete anything there from these queries and will comment later today.

I'm not sure why the check is there otherwise 🤷 Maybe we removed something around that afterwards.

Reply:
@sgiehl looking at the logs none of the accounts is purging any data there. I also went through all `UPDATE` statements etc. I cannot find anything that would create or update empty `ts_archived`. 

What's your thoughts on it? Keep the where clause anywhere or good to merge? or maybe I should even remove the `getPlaceholderArchiveIds` logic as it doesn't seem to be needed?

Reply:
Ok. That fits my findings when looking through the code. In that case we maybe could also directly remove `getPlaceholderArchiveIds`, as there shouldn't be any sense in having it.
And we maybe could also add an according `DELETE` query for every archive table in the update script, to ensure non of such archives exists anymore after updating to 5.0. Just in case there might be any if someone updates from an older version.

Reply:
Good idea 👍 Thanks for that. Just made that change @sgiehl